src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -57,13 +132,146 @@ public boolean equals(Object other) {
         }
 
         DeleteCommand otherDeleteCommand = (DeleteCommand) other;
-        return targetIndex.equals(otherDeleteCommand.targetIndex);
+
+        return Objects.equals(nric, otherDeleteCommand.nric)
+                && Objects.equals(name, otherDeleteCommand.name)
+                && Objects.equals(deletePersonDescriptor, otherDeleteCommand.deletePersonDescriptor);
     }
 
     @Override
     public String toString() {
         return new ToStringBuilder(this)
-                .add("targetIndex", targetIndex)
+                .add("nric", nric)
+                .add("name", name)
+                .add("deletePersonDescriptor", deletePersonDescriptor)
                 .toString();
     }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code deletePersonDescriptor}.
+     */
+    public static Person createEditedPerson(Person personToEdit, DeletePersonDescriptor deletePersonDescriptor) {
```

Comment:
Perhaps it may be prudent to change the name of this method to be more related to `Delete` rather than `Edit`, since `Edit` is a command that exists in the set of `Command`s? This is purely cosmetic however, so change this at your discretion.